packages/stencil/components/code/src/components/osds-code/osds-code.scss

```diff
@@ -9,24 +9,34 @@
 // CSS for the component but not for theming
 :host {
   @include osds-code-on-main-element() {
-    position: relative;
-    margin: 0;
-    display: flex;
-    justify-content: space-between;
+    display: grid;
+    grid-template-columns: 1fr auto;
+    grid-template-rows: 1fr;
+    gap: 0;
+    grid-template-areas:
+    ". .";
+    justify-content: start;
+    align-content: start;
+    grid-auto-columns: 100%;
```

Comment:
Could be a nice opportunity to normalize and simplify its style

you don't have to put all thos property, container would actually only needs:
```
display: grid;
grid-template-columns: 1fr auto;
grid-column-gap: var(--ods-size-code-md-padding);
```
the only issue is that this will add a gap even if code is not display, so `flex` is easier here:
```
display: flex;
flex-flow: row;
grid-column-gap: var(--ods-size-code-md-padding);
```
(and remove `width: 100%;`)

Then remove padding from both children.

This way padding is consistent across the whole component without defining any on child elements;
WDYT?